testing/deployer/sprawl/catalog.go

```diff
@@ -318,13 +318,45 @@ func serviceToCatalogRegistration(
 			Address: node.LocalAddress(),
 		},
 	}
+	if svc.IsMeshGateway {
+		reg.Service.Kind = api.ServiceKindMeshGateway
+		reg.Service.Proxy = &api.AgentServiceConnectProxyConfig{
+			Config: map[string]interface{}{
+				"envoy_gateway_no_default_bind":       true,
```

Comment:
TBH I don't know what these do or if they're necessary, but `consul connect envoy -mesh-gateway` set them on registration.

Reply:
The no default bind one will not bind to the main service address and the bind tagged addrs one then instructs it to bind envoy listeners to all the tagged addrs instead.